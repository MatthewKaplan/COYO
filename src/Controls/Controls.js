import React, { Component } from 'react';
import '../Styles/_Controls.scss';

class Controls extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  displayAllStudios = (e) => {
    e.preventDefault()
    this.props.storeRendered(this.props.studios)
  }

  priceFilter = () => {
    let ordered;
    if(this.props.rendered.length >= 1) {
      ordered = this.props.rendered.sort((a, b) => (a.dropInFee > b.dropInFee) ? 1 : -1);
    } else {
      ordered = this.props.studios.sort((a, b) => (a.dropInFee > b.dropInFee) ? 1 : -1);
    }

    ordered.forEach(el => {
      if(el.dropInFee.endsWith('Suggested donation')) {
        let lastElement = ordered.pop();
        ordered.unshift(lastElement);
      }
    });
    return ordered;
  }

  sortByPrice = (e) => {
    e.preventDefault()
    let filteredPrice = this.priceFilter();
    this.props.storeRendered(filteredPrice);
  }

  render() {
    return  (
      <section className='btnContainer'>
          <input 
          value='Show all studios'
          type='submit'
          className='showAll'
          onClick={this.displayAllStudios}
          data-test="show-all-button"
          />
          <input
          value='Sort by price' 
          type='submit'
          className='sort'
          onClick={this.sortByPrice}
          data-test="sort-price-button"
          />
      </section>
    )
  }
}

export default Controls;
