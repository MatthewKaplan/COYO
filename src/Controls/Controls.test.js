import React from "react";
import Controls from "./Controls";
import { shallow } from "enzyme";
import MockData from "../Helper/mockData";

let mockRendered = MockData.mockStudios;

describe("Controls", () => {
  let wrapper, instance;

  let mockStudio = [
    {
      name: "Whole Yoga",
      mainTypeOffered: 5,
      address: "1735 E 17th Ave, Denver, CO 80218",
      phoneNum: "(303)333-9642",
      openEveryday: true,
      mission: "Where body and mind become whole.",
      googleRating: 4.8,
      dropInFee: "$17.00",
      teacherTraining: true,
      link: "https://wholeyoga.com",
      img: "https://i.imgur.com/w7KfQWv.jpg"
    },
    {
      name: "Rino Yoga Social",
      mainTypeOffered: 8,
      address: "3101 Walnut St, Denver, CO 80205",
      phoneNum: "(720)724-8393",
      openEveryday: true,
      mission: "At RYS, our mission is simple: make yoga accessible for all.",
      googleRating: 5,
      dropInFee: "Suggested donation",
      teacherTraining: false,
      link: "https://www.rinoyogasocial.com/",
      img: "https://i.imgur.com/nKIbWP2.jpg"
    },
    {
      name: "Denver MySore",
      mainTypeOffered: 4,
      address: "910 Santa Fe Dr, Denver, CO 80204",
      phoneNum: "(303)477-0370",
      openEveryday: false,
      mission: "Classes in the heart of Denver's Santa Fe Art District.",
      googleRating: 5,
      dropInFee: "$20.00",
      teacherTraining: false,
      link:
        "http://www.denvermysore.com/ashtanga-yoga-and-the-denver-mysore-community/practice-ashtanga-yoga/",
      img: "https://i.imgur.com/esQTB37.jpg"
    }
  ];

  const mockStoreEmptyType = jest.fn();
  const mockStoreRendered = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Controls
        storeEmptyType={mockStoreEmptyType}
        studios={mockStudio}
        storeRendered={mockStoreRendered}
        rendered={mockRendered}
      />
    );
    instance = wrapper.instance();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("displayAllStudios", () => {
    it("should render a show all button", () => {
      let MockFn = jest.spyOn(instance, "displayAllStudios");
      wrapper.find("[data-test='show-all-button']").simulate("click");
      expect(MockFn).toHaveBeenCalled();
    });

    it("should invoke 'storeEmptyType' with the correct params", () => {
      instance.displayAllStudios();
      expect(mockStoreEmptyType).toHaveBeenCalledWith("");
    });

    it("should invoke 'storeRendered' with the correct params", () => {
      instance.displayAllStudios();
      expect(mockStoreRendered).toHaveBeenCalledWith(mockStudio);
    });
  });

  describe("sortByPrice", () => {
    it("should render a sort by price button", () => {
      let MockFn = jest.spyOn(instance, "sortByPrice");
      wrapper.find("[data-test='sort-price-button']").simulate("click");
      expect(MockFn).toHaveBeenCalled();
    });

    it("should invoke 'priceFilter'", () => {
      let MockFn = jest.spyOn(instance, "priceFilter");
      instance.sortByPrice();
      expect(MockFn).toHaveBeenCalled();
    });

    it("should invoke 'storeRendered' with the correct params", () => {
      instance.sortByPrice();
      expect(mockStoreRendered).toHaveBeenCalledWith(mockStudio);
    });
  });

  describe("priceFilter", () => {
    it("should return proper value if rendered length is greater or equal to one", () => {
      const results = instance.priceFilter();
      expect(results).toEqual(mockRendered);
    });

    it("should return proper value if rendered length is less than one", () => {
      mockRendered = [];
      wrapper = shallow(
        <Controls
          storeEmptyType={mockStoreEmptyType}
          studios={mockStudio}
          storeRendered={mockStoreRendered}
          rendered={mockRendered}
        />
      );
      instance = wrapper.instance();
      const results = instance.priceFilter();
      expect(results).toEqual(mockStudio);
    });
  });
});
