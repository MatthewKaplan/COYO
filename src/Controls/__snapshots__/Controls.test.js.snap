// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="showAll"
          onClick={[Function]}
          type="submit"
          value="Show all studios"
        />,
        <input
          className="sort"
          onClick={[Function]}
          type="submit"
          value="Sort by price"
        />,
      ],
      "className": "btnContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "showAll",
          "onClick": [Function],
          "type": "submit",
          "value": "Show all studios",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "sort",
          "onClick": [Function],
          "type": "submit",
          "value": "Sort by price",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="showAll"
            onClick={[Function]}
            type="submit"
            value="Show all studios"
          />,
          <input
            className="sort"
            onClick={[Function]}
            type="submit"
            value="Sort by price"
          />,
        ],
        "className": "btnContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "showAll",
            "onClick": [Function],
            "type": "submit",
            "value": "Show all studios",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sort",
            "onClick": [Function],
            "type": "submit",
            "value": "Sort by price",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
